.class Lcom/virgilsecurity/pythia/PythiaContext$Builder$1;
.super Ljava/lang/Object;
.source "PythiaContext.java"

# interfaces
.implements Lcom/virgilsecurity/sdk/jwt/accessProviders/CachingJwtProvider$RenewJwtCallback;


# annotations
.annotation system Ldalvik/annotation/EnclosingMethod;
    value = Lcom/virgilsecurity/pythia/PythiaContext$Builder;->build()Lcom/virgilsecurity/pythia/PythiaContext;
.end annotation

.annotation system Ldalvik/annotation/InnerClass;
    accessFlags = 0x0
    name = null
.end annotation


# instance fields
.field final synthetic this$0:Lcom/virgilsecurity/pythia/PythiaContext$Builder;

.field final synthetic val$generator:Lcom/virgilsecurity/sdk/jwt/JwtGenerator;


# direct methods
.method constructor <init>(Lcom/virgilsecurity/pythia/PythiaContext$Builder;Lcom/virgilsecurity/sdk/jwt/JwtGenerator;)V
    .locals 0
    .param p1, "this$0"    # Lcom/virgilsecurity/pythia/PythiaContext$Builder;

    .line 172
    iput-object p1, p0, Lcom/virgilsecurity/pythia/PythiaContext$Builder$1;->this$0:Lcom/virgilsecurity/pythia/PythiaContext$Builder;

    iput-object p2, p0, Lcom/virgilsecurity/pythia/PythiaContext$Builder$1;->val$generator:Lcom/virgilsecurity/sdk/jwt/JwtGenerator;

    invoke-direct {p0}, Ljava/lang/Object;-><init>()V

    return-void
.end method


# virtual methods
.method public renewJwt(Lcom/virgilsecurity/sdk/jwt/TokenContext;)Lcom/virgilsecurity/sdk/jwt/Jwt;
    .locals 4
    .param p1, "tokenContext"    # Lcom/virgilsecurity/sdk/jwt/TokenContext;

    .line 177
    :try_start_0
    iget-object v0, p0, Lcom/virgilsecurity/pythia/PythiaContext$Builder$1;->val$generator:Lcom/virgilsecurity/sdk/jwt/JwtGenerator;

    const-string v1, "PYTHIA-CLIENT"

    invoke-virtual {v0, v1}, Lcom/virgilsecurity/sdk/jwt/JwtGenerator;->generateToken(Ljava/lang/String;)Lcom/virgilsecurity/sdk/jwt/Jwt;

    move-result-object v0
    :try_end_0
    .catch Lcom/virgilsecurity/sdk/crypto/exceptions/CryptoException; {:try_start_0 .. :try_end_0} :catch_0

    return-object v0

    .line 178
    :catch_0
    move-exception v0

    .line 180
    .local v0, "e":Lcom/virgilsecurity/sdk/crypto/exceptions/CryptoException;
    invoke-static {}, Lcom/virgilsecurity/pythia/PythiaContext;->access$000()Ljava/util/logging/Logger;

    move-result-object v1

    sget-object v2, Ljava/util/logging/Level;->SEVERE:Ljava/util/logging/Level;

    const-string v3, "Jwt token couldn\'t be generated"

    invoke-virtual {v1, v2, v3, v0}, Ljava/util/logging/Logger;->log(Ljava/util/logging/Level;Ljava/lang/String;Ljava/lang/Throwable;)V

    .line 182
    .end local v0    # "e":Lcom/virgilsecurity/sdk/crypto/exceptions/CryptoException;
    const/4 v0, 0x0

    return-object v0
.end method
